pkg/scheduler/factory.go

```diff
@@ -307,12 +308,61 @@ func (c *Configurator) createFromPolicy(policy schedulerapi.Policy) (*Scheduler,
 		},
 	}
 
-	// Use the latest version to set the default values for plugin configs.
-	v1beta2.ConvertToInternalPluginConfigArgs(&c.profiles[0])
+	if err := defaultPluginConfigArgs(&c.profiles[0]); err != nil {
+		return nil, err
+	}
 
 	return c.create()
 }
 
+func defaultPluginConfigArgs(prof *schedulerapi.KubeSchedulerProfile) error {
+	scheme := v1beta2.GetPluginArgConversionScheme()
+	existingConfigs := sets.NewString()
+	for j := range prof.PluginConfig {
+		existingConfigs.Insert(prof.PluginConfig[j].Name)
+		args := prof.PluginConfig[j].Args
+		if args == nil {
+			continue
+		}
+		if _, isUnknown := args.(*runtime.Unknown); isUnknown {
+			continue
+		}
+		scheme.Default(args)
```

Comment:
Are you sure this works? The types produced by the legacy registry are already internal. Thus, I think this and the line below do nothing.

Reply:
yeah, this doesn't, but the one for the newly added ones below does. but this was the existing behavior in the now removed framework.go defaulting logic (i.e., do nothing for existing plugin configs). In any case, I think we should be able to remove this completely in the next release by removing the policy flag.

Reply:
removed it and added a comment.